packages/internal/src/utils.js

```diff
@@ -348,3 +349,101 @@ export const allValues = async obj => {
   // @ts-expect-error cast
   return harden(fromEntries(zip(keys(obj), resolved)));
 };
+
+/**
+ * A tee implementation where all readers are synchronized with each other.
+ * They all consume the source stream in lockstep, and any one returning or
+ * throwing early will affect the others.
+ *
+ * @template [T=unknown]
+ * @param {AsyncIterator<T, void, void>} stream
+ * @param {number} quantity
+ */
+export const syncTee = (stream, quantity) => {
```

Comment:
Given that this is very much about *async* iterators, maybe name it `synchronizedTee` or `lockstepTee`?

Also, maybe `numReaders` or `readerCount` instead of `quantity`, to make the docstring a bit more instructive.